### PR TITLE
Add async counter example

### PR DESCRIPTION
**Pull request by sotojuan**
This is a common example in React/Redux world so it'd be nice to have it for comparison's sake.

In the future I'd like to be able to "cancel" an async increment/decrement such that if the async increment button is clicked, then the cancel button is clicked before the timeout ends, `send('increment', done)` is never called and nothing happens.

### CODE REVIEW

File: examples/async-counter/client.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  state: {
+    counter: 0
+  },
+  reducers: {
+    increment: (data, state) => ({ counter: state.counter + 1 }),
+    decrement: (data, state) => ({ counter: state.counter - 1 })
+  },
+  effects: {
+    incrementAsync: function (data, state, send, done) {
+      setTimeout(() => send('increment', done), 1000)
```
Comment (timwis) [inline examples/async-counter/client.js]: Hey I think you meant to put `done` as the third param here. The second param is for `data`, which can be `null` in this case.

Comment (yoshuawuyts) [inline examples/async-counter/client.js]: This is totally valid too; `data` is optional so is set to `null` if not passed explicitely.

[Ref in `barracks` codebase](https://github.com/yoshuawuyts/barracks/blob/master/index.js#L108-L111)

**Comment by yoshuawuyts** [comment]
LGTM

**Comment by sotojuan** [comment]
Thanks, @yoshuawuyts feel free to squash and merge when you can :-)

**Comment by yoshuawuyts** [comment]
@sotojuan looks like with the `master` branch guards, I can't squash and merge - could you squash the commits and force push so I can merge? Thanks! :grin:

**Comment by sotojuan** [comment]
Hey @yoshuawuyts not sure if I did it correctly—I don't mind just closing this PR and submitting another one lol. Really bad at Git.

**Comment by sotojuan** [comment]
@yoshuawuyts Let me know if you can merge ASAP because IDK what I am doing w/ the Git history/rebase anymore ha. Otherwise I don't mind closing and redoing a PR again.

**Comment by yoshuawuyts** [comment]
hah cool yeah let's merge (':

**Pull request merged**